motioneye/motionctl.py

```diff
@@ -186,6 +187,26 @@ def stop(invalidate=False):
             if e.errno not in (errno.ESRCH, errno.ECHILD):
                 raise
 
+            # even if motion is killed, the port is not always free
+            # so we wait until we are able to bind it again
+            s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            max_wait = 30
+            waited_for = 0
+            while waited_for <= max_wait:
+                try:
+                    s.bind(('127.0.0.1', settings.MOTION_CONTROL_PORT))
+                    s.close()
+                    logging.debug('motion port has been released')
+                    return
+
+                except OSError as e:
+                    logging.debug(f'motion port has not been released yet: {e}')
```

Comment:
Nits!
```suggestion
                    logging.debug(f'motion port has not yet been released: {e!r}')
```

Reply:
Checked str vs repr variants of exception messages, and indeed we should always return the repr value 👍.